release-content/0.15/release-notes/14791_Required_Components.md

```diff
@@ -1,4 +1,421 @@
+
+```rust
+#[derive(Component)]
+#[require(Team(blue_team)]
+struct Player;
+
+fn blue_team() -> Team {
+    Team::Blue
+}
+```
+
+To save space, you can also pass a closure to the require directly:
+
+```rust
+#[derive(Component)]
+#[require(Team(|| Team::Blue))]
+struct Player;
+```
+
+### Isn't this a bit like inheritance?
+
+**Required Components** _can_ be considered a form of inheritance. But it is notably _not_ traditional object-oriented inheritance. Instead it is "inheritance by composition". A `Button` widget can (and should) require `Node` to make it a "UI node". In a way, a `Button` "is a" `Node`. But unlike traditional inheritance:
+
+1. `Button` and `Node` are still two entirely separate types (with their own data), which you query for separately in the ECS.
+2. A `Button` can require more components in _addition_ to `Node`. You aren't limited to "straight line" standard object-oriented inheritance. Composition is still the dominating pattern.
+3. You don't _need_ to require components to add them. You can still tack on whatever additional components you want during spawn to add behaviors in the normal "composition style".
+
+### What is happening to Bundles?
+
+The `Bundle` trait will continue to exist, and it is still the fundamental building block for insert APIs (tuples of components still implement `Bundle`). Developers are still free to define their own custom bundles using the `Bundle` derive. Bundles play nicely with **Required Components**, so you can use them with each other.
+
+That being said, as of Bevy **0.15** we have deprecated all built in bundles like `SpriteBundle`, `NodeBundle`, `PbrBundle`, etc in favor of **Required Components**. In general, **Required Components** are now the preferred / idiomatic approach. We encourage Bevy plugin and app developers to port their bundles over to **Required Components**.
+
+### Porting Bevy to Required Components
+
+As mentioned above, _all_ built in Bevy bundles have been deprecated in favor of **Required Components**. We've also made API changes to take advantage of this new paradigm. This does mean breakage in a few places, but the changes are so nice that we think people won't complain too much :)
+
+In general, we are moving in the direction specified by our [Next Generation Scene / UI](https://github.com/bevyengine/bevy/discussions/14437) document. Some general design guidelines:
+
+1. When spawning an entity, generally there should be a "driving concept" component.  When implementing a new entity type / behavior, give it a concept name ... that is the name of your "driving component" (ex: the "player" concept is a `Player` component). That component should require any additional components necessary to perform its functionality.
+2. People should think directly in terms of components and their fields when spawning. Prefer using component fields directly on the "concept component" as the "public API" for the feature.
```

Comment:
This wording here makes it sound like the `Player` example from above should have a `team` field. I think part of what makes the explanation awkward is that the previous  `Player` component (and also the `Team` component) doesn't have any fields, so the reader is forced to come up with their own example that fits the explanation while reading, and that explanation might lead them directly into wondering why `team` is a required component and not a field above.

This might be as easy as introducing a field on `Player` for the example so that the concept is seen before this paragraph (maybe `display_name`?).

Reply:
Yeah that makes sense.

> This might be as easy as introducing a field on Player 

Yeah I think thats a reasonable solve.

The field vs required component debate is nuanced to the point that I'd rather not address it head on in this section. In a way, it is already covered in the Porting Bevy to Required Components in the "design guidelines" section. I'd prefer to not go any deeper than that.